applications/crossbar/src/modules/cb_configs.erl

```diff
@@ -15,198 +12,164 @@
-validate(Context, Config) ->
-    validate(Context, cb_context:req_verb(Context), Config).
+validate(Context, ConfigId) ->
+    try
+        validate(Context, cb_context:req_verb(Context), ConfigId)
+    catch
+        _:{badmatch, {invalid_document, Errors} = Error} ->
+            lager:error("schema validation error: ~p", [parse_error(Error)]),
+            cb_context:failed(Context, Errors);
+        _:{badmatch,{error,enoent}} ->
+            lager:error("schema validation error: no schema for config"),
+            cb_context:add_system_error(bad_identifier, Context);
+        _:{badmatch,{error,not_found}} ->
+            lager:error("schema validation error: no schema for config"),
+            cb_context:add_system_error(bad_identifier, Context);
+        _C:E ->
+            lager:error("validation generic error: ~p", [E]),
+            ST = erlang:get_stacktrace(),
+            kz_util:log_stacktrace(ST),
+            error_validation(Context)
+    end.
 
 -spec validate(cb_context:context(), http_method(), path_token()) -> cb_context:context().
-validate(Context, ?HTTP_GET, Config) -> read(Config, Context);
-validate(Context, ?HTTP_PUT, Config) -> create(Config, Context);
-validate(Context, ?HTTP_POST, Config) -> update(Config, Context);
-validate(Context, ?HTTP_PATCH, Config) -> validate_patch(Config, Context);
-validate(Context, ?HTTP_DELETE, Config) -> read(Config, Context).
-
-%%--------------------------------------------------------------------
-%% @public
-%% @doc
-%% If the HTTP verb is a GET, execute necessary code to fulfill the GET
-%% request. Generally, this will involve stripping pvt fields and loading
-%% the resource into the resp_data, resp_headers, etc...
-%% @end
-%%--------------------------------------------------------------------
--spec get(cb_context:context(), path_token()) -> cb_context:context().
-get(Context, _) ->
-    Context.
 
-%%--------------------------------------------------------------------
-%% @public
-%% @doc
-%% If the HTTP verib is PUT, execute the actual action, usually a db save.
-%% @end
-%%--------------------------------------------------------------------
+validate(Context, ?HTTP_GET, ConfigId) ->
+    JObj = kapps_config_util:get_config(cb_context:account_id(Context), ConfigId),
+    crossbar_doc:handle_datamgr_success(set_id(ConfigId, JObj), Context);
+
+validate(Context, ?HTTP_DELETE, ConfigId) ->
+    Doc = case kapps_config_util:load_config_from_account(cb_context:account_id(Context), ConfigId) of
+              {ok, Document} -> Document;
+              _ -> set_id(ConfigId, kz_json:new())
+          end,
+    pass_validation(Context, Doc);
+
+validate(Context, ?HTTP_PUT, ConfigId) ->
+    validate(Context, ?HTTP_POST, ConfigId);
+
+validate(Context, ?HTTP_PATCH, ConfigId) ->
+    Parent = kapps_config_util:get_config(cb_context:account_id(Context), ConfigId),
+    save_diff(validate_diff(Context, ConfigId, Parent), ConfigId, Parent);
```

Comment:
maybe because of the way it was validated. were you calling `crossbar_doc:handle_datamgr_success` like in validate for get ? check the other modules that implement patch.